pyro/poutine/trace.py

```diff
@@ -1,8 +1,19 @@
 from __future__ import absolute_import, division, print_function
 
 import collections
+import warnings
 
 import networkx
+import numpy as np
+
+
+def _warn_if_nan(name, variable):
+    value = variable.data[0]
```

Comment:
This seems preferable so as to avoid large tensor checks. I suppose this will leave out some small cases where the NaNs/Infs are in different locations and don't propagate across the tensor?

Reply:
We're only checking on `log_pdf` which has a single element. The reduction is already done by `batch_log_pdf()` via `.sum()`, so we do indeed check for NANs everywhere.